src/bindgen/bindings.rs

```diff
@@ -57,25 +62,32 @@ impl Bindings {
         any
     }
 
-    pub fn struct_field_names(&self, path: &BindgenPath) -> Option<Vec<String>> {
-        let item_container = self.struct_map.get_items(path)?;
-        let mut fields = Vec::<String>::new();
-        for container in item_container {
-            let mut pos: usize = 0;
-            if let ItemContainer::Struct(ref st) = container {
-                for field in &st.fields {
-                    if let Some(found_pos) = fields.iter().position(|v| *v == field.0) {
-                        pos = found_pos + 1;
-                    } else {
-                        fields.insert(pos, field.0.clone());
-                        pos += 1;
+    pub fn struct_field_names(&self, path: &BindgenPath) -> Option<Rc<Vec<String>>> {
+        let mut memos = self.struct_fileds_memo.borrow_mut();
+        if let Some(memo) = memos.get(path) {
+            Some(memo.clone())
+        } else {
+            let item_container = self.struct_map.get_items(path)?;
+            let mut fields = Vec::<String>::new();
+            for container in item_container {
+                let mut pos: usize = 0;
+                if let ItemContainer::Struct(ref st) = container {
+                    for field in &st.fields {
+                        if let Some(found_pos) = fields.iter().position(|v| *v == field.0) {
+                            pos = found_pos + 1;
+                        } else {
+                            fields.insert(pos, field.0.clone());
+                            pos += 1;
+                        }
                     }
+                } else {
+                    unreachable!()
                 }
-            } else {
-                unreachable!()
             }
+
+            memos.insert(path.clone(), Rc::new(fields));
+            memos.get(path).map(|val| val.clone())
```

Comment:
This always returns `Some()` right?

This should probably do something like:

```rust
let fields = Rc::new(fields);
memos.insert(path.clone(), fields.clone());
fields
```

And just return `Rc<Vec<..>>`.